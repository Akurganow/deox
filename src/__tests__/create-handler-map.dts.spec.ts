import { createAction } from '../create-action'
import { createHandlerMap as handle } from '../create-handler-map'

// @dts-jest:group createHandlerMap

const increment = createAction('[Counter] increment')
const increase = createAction('[Counter] increase')

// @dts-jest:pass:snap
handle(increment, (state: number) => state + 1)

// @dts-jest:pass:snap
handle([increment, increase], (state: number) => state + 1)
