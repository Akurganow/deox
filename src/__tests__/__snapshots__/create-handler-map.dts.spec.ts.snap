// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createHandlerMap handle([increment, increase], (state: number) => state + 1) (type) should match snapshot: createHandlerMap handle([increment, increase], (state: number) => state + 1) 1`] = `"import(\\"/Users/brodmann/Projects/@thebrodmann/deox/src/create-handler-map\\").HandlerMap<number, { type: \\"[Counter] increment\\"; } | { type: \\"[Counter] increase\\"; }>"`;

exports[`createHandlerMap handle(increment, (state: number) => state + 1) (type) should match snapshot: createHandlerMap handle(increment, (state: number) => state + 1) 1`] = `"import(\\"/Users/brodmann/Projects/@thebrodmann/deox/src/create-handler-map\\").HandlerMap<number, { type: \\"[Counter] increment\\"; }>"`;
